src/Symfony/Component/VarDumper/Caster/SocketCaster.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+class SocketCaster
+{
+    public static function castSocket(\Socket $h, array $a, Stub $stub, bool $isNested): array
+    {
+        socket_getsockname($h, $addr, $port);
+        $info = stream_get_meta_data(socket_export_stream($h));
+
+        $a += [
+            Caster::PREFIX_VIRTUAL.'address' => $addr,
+            Caster::PREFIX_VIRTUAL.'port' => $port,
+            Caster::PREFIX_VIRTUAL.'info' => new EnumStub(array_intersect_key(
```

Comment:
Once you remove all the non-sense ConstStub, you can also remove the `array_intersect_key` entirely, using `$info` directly as value for the info property.